### ISSUE TITLE
Passive Endermen causes errors and prevents Endermen from spawning

### ISSUE THREAD

**Issue by Coyote21** [ricksouth/serilum-mc-mods#1075]
## **Information**
Minecraft version: 1.18.2
Modloader: Fabric
Fabric loader version: 0.14.6
Environment: Multiplayer

Mod name: passiveendermen-fabric_1.18.x-2.3.jar
Mod version: 2.3

## **Description**
I no longer see any Endermen spawn in the overworld

## ** Log Excerpt **

[Server thread/WARN]: Failed to create mob
java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.util.HashMap.get(Object)" is null
	at Not Enough Crashes deobfuscated stack trace.(1.18.2+build.3) ~[?:?]
	at net.minecraft.entity.mob.EndermanEntity.<init>(EndermanEntity:89) ~[?:?]
	at net.minecraft.world.SpawnHelper.createMob(SpawnHelper:354) ~[?:?]
	at net.minecraft.entity.EntityType.create(EntityType:478) ~[?:?]
	at net.minecraft.world.SpawnHelper.spawnEntitiesInChunk(SpawnHelper:283) ~[?:?]
	at net.minecraft.world.SpawnHelper.spawnEntitiesInChunk(SpawnHelper:211) ~[?:?]
	at net.minecraft.world.SpawnHelper.spawn(SpawnHelper:199) ~[?:?]
	at net.minecraft.server.world.ServerChunkManager.tickChunks(ServerChunkManager:375) ~[?:?]
	at net.minecraft.server.world.ServerChunkManager.tick(ServerChunkManager:322) ~[?:?]
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld:311) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer:907) ~[server-intermediary.jar:?]
	at net.minecraft.server.dedicated.MinecraftDedicatedServer.tickWorlds(MinecraftDedicatedServer:322) ~[?:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer:851) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer:697) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer:273) ~[server-intermediary.jar:?]

**Comment by skcipherhash** [comment]
Can confirm. Enderman spawn eggs doesn't work either.
<details>
<summary>Log:</summary>
<br>

```
[22:45:23] [Server thread/ERROR]: Failed to handle packet net.minecraft.class_2885@742d7f42, suppressing error
java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because the return value of "java.util.HashMap.get(Object)" is null
	at net.minecraft.class_1560.<init>(class_1560.java:89) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_5883(class_1299.java:487) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_5888(class_1299.java:356) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_5899(class_1299.java:344) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_5894(class_1299.java:335) ~[client-intermediary.jar:?]
	at net.minecraft.class_1826.method_7884(class_1826.java:90) ~[client-intermediary.jar:?]
	at net.minecraft.class_1799.method_7981(class_1799.java:264) ~[client-intermediary.jar:?]
	at net.minecraft.class_3225.method_14262(class_3225.java:355) ~[client-intermediary.jar:?]
	at net.minecraft.class_3244.method_12046(class_3244.java:1115) ~[client-intermediary.jar:?]
	at net.minecraft.class_2885.method_12547(class_2885.java:34) ~[client-intermediary.jar:?]
	at net.minecraft.class_2885.method_11054(class_2885.java:8) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:22) ~[client-intermediary.jar:?]
	at net.minecraft.class_3738.run(class_3738.java:18) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:157) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_24306(MinecraftServer.java:776) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_18859(MinecraftServer.java:160) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:758) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:752) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18857(class_1255.java:140) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:737) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:669) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:258) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```
</details>

Minecraft 1.19 with Fabric Loader 0.14.8, passiveendermen-fabric_1.19.x-2.4

**Comment by ricksouth** [comment]
I have just released version 3.0 which fixes the multiple things that was causing this. Thank you for the patience, letting me know and giving the mod a try. I appreciate it.

https://www.curseforge.com/minecraft/mc-mods/passive-endermen-fabric/files
